utils/clickhouse.h

```diff
@@ -0,0 +1,134 @@
+/**
+ *		Tempesta FW
+ *
+ * Copyright (C) 2024 Tempesta Technologies, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License,
+ * or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59
+ * Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ */
+
+#ifndef __TFW_CLICKHOUSE_H__
+#define __TFW_CLICKHOUSE_H__
+
+#include <iostream>
+#include <clickhouse/client.h>
+#include <clickhouse/base/socket.h>
+
+using namespace clickhouse;
+using namespace std;
+
+/**
+ * Class for sending records to a Clickhouse database.
+ * It manages multiple instances of the Clickhouse client (one per thread) to
+ * handle concurrent operations, as the Clickhouse client itself is not
+ * thread-safe.
```

Comment:
In the current architecture, you have only one instance of `TfwMmapBufferReader` and one instance of `TfwClickhouse` and you do the per-cpu logic internally inside the objects. This logic is very fragile: once you get a new data member of any of the class, you need to keep in mind synchronization. Also `cpu` is passed as an argument of many methods.

It's better to make the classes simpler: one is to manage local Clickhouse client and another for Tempesta log. Each thread object (aka worker) has relation to `std::thread` and it's own instance for the Clickhouse and Tempesta log handlers. The most of the logic will not need to care about CPUs and any synchronization.

Reply:
I wanted to store all the code connected to work with ring-buffer in one place. We need to calculate buffers offsets in the mapped memory, work with common file descriptor. I would prefer to have it in one class.
Agree that TfwClickhouse could be simplified.

We could leave TfwMmapBufferReader as is, create an instance of this class, create per-cpu instances of TfwClickhouse and set callbacks. Maybe we can use a private_data field in the TfwThread to store a reference to a TfwClickhouse instance.

Reply:
Made both of the classes single-threaded.